riak/transports/pbc/connection.py

```diff
@@ -84,7 +84,10 @@ def _auth(self):
         """
         req = riak.pb.riak_pb2.RpbAuthReq()
         req.user = str_to_bytes(self._client._credentials.username)
-        req.password = str_to_bytes(self._client._credentials.password)
+        password = self._client._credentials.password
+        if not password:
+            password = ''
+        req.password = str_to_bytes(password)
```

Comment:
Even though password is not required by cert-based auth, it is required in the PB message, which is why an empty string is encoded here.